src/collective/listingviews/browser/views/listing_view.py

```diff
@@ -22,27 +20,28 @@ class ListingView(BrowserView):
     security = ClassSecurityInfo()
     security.declareObjectProtected(Permissions.view)
 
-    def __init__(self, context, request):
+    def __init__(self, context, request, name):
         super(ListingView, self).__init__(context, request)
-        self.listing_view_adapter = getMultiAdapter((context,request), name='listing_view_adapter')
+        self.__adapter_name__ = name
+
+        self.listing_view_adapter = getMultiAdapter((self.context, self.request), name='listing_view_adapter')
 
         if IFacetedLayout is not None and \
-            (IFacetedSearchMode.providedBy(self.context) or IFacetedNavigable.providedBy(self.context)):
+                (IFacetedSearchMode.providedBy(self.context) or IFacetedNavigable.providedBy(self.context)):
             # Case: It's being used from facetednavigation
             self.set_listing_view(getListingNameFromView(IFacetedLayout(self.context).layout))
         else:
-            view_name = getAdapterName()
-            self.listing_view_adapter.set_listing_view(getListingNameFromView(view_name))
-            # # Case: It's being used from a normal display menu view
-            # view_name = request.getURL().split('/')[-1]
-            # self.set_listing_view(getListingNameFromView(view_name))
+            view_name = getListingNameFromView(self.__adapter_name__)
+            self.listing_view_adapter.set_listing_view(view_name)
 
     def render(self):
         return self.index()
 
     security.declareProtected(Permissions.view, '__call__')
     def __call__(self):
         """ Render the view """
+
```

Comment:
should probably remove these blank lines, they don't seem to be doing anything but cluttering the pr